core-primitives/stf-executor/src/executor.rs

```diff
@@ -186,10 +189,27 @@ where
 			let (state_lock, mut state) = self.state_handler.load_for_mutation(&shard_id)?;
 			match Stf::update_parentchain_block(&mut state, header.clone()) {
 				Ok(_) => {
-					self.state_handler.write_after_mutation(state, state_lock, &shard_id)?;
+					//self.state_handler.write_after_mutation(state, state_lock, &shard_id)?;
 				},
 				Err(e) => error!("Could not update parentchain block. {:?}: {:?}", shard_id, e),
 			}
+
+			if let Some(maybe_next_ceremony_phase) =
```

Comment:
Since you are adding new logic here, can you adapt the comment above `// Update parentchain block on all states.`? Mby something like `Update shard states`